sasmodels/modelinfo.py

```diff
@@ -978,6 +969,32 @@ def make_model_info(kernel_module):
     if callable(info.Iq) and parameters.has_2d:
         raise ValueError("oriented python models not supported")
 
+    # CRUFT: support old-style ER() for effective radius
+    # radius_effective(mode, ...), if it is present, should be a callable
+    # python function accepting the shape parameters, and have a corresponding
+    # Iq function in python (Fq in python is not suported at this time).
+    # If ER() is present, it might represent a 4.2 model that hasn't been
+    # converted, or it might represent a 4.2+5.0 model with both ER and
+    # radius_effective defined. Unfortunately, we can't easily tell whether
+    # radius_effective is defined in the C sources without loading them,
+    # so just issue a warning if we see ER in a C model.
+    ER = getattr(kernel_module, 'ER', None)
+    if ER is not None:
+        if callable(info.Iq) and info.radius_effective is None:
```

Comment:
A deprecation warning would be useful here so users know to update their models before end of life is reached.

Reply:
I changed it then backed off.  It is okay to have the warning as part of the 4.3 release but not any 5.x point releases that happen in the meantime. We don't want to have a deprecation before the new functionality is available in the 4.x branch, or the 4.x branch is officially abandoned.